### PR TITLE
[Security] Upgrade all the things

### PR DESCRIPTION
**Pull request by eatyourgreens**
Staging branch URL: https://pr-252.lab-preview.zooniverse.org/

Closes https://github.com/zooniverse/how-to-zooniverse/issues/137.

- Use [`npm ci`](https://docs.npmjs.com/cli/v7/commands/npm-ci) for builds.
- Re-use the npm cache across installs, when `package-lock.json` is unchanged.
- Add `$HEAD_COMMIT` to the deployment environment on the main branch. It's used to add the latest commit hash to the page HTML.
- Publish `/commit_id.txt` on deploy.
- Run `npm audit fix` to fix some outdated dependencies.
- Upgrade `mocha` to 8.4 to fix a security warning for old dependencies. Replace `mocha.opts` with `.mocharc.js`.
- Use `https:` to install Zooniverse-React-Components. `npm ci` breaks with `ssh:` URLs. See https://github.com/actions/setup-node/issues/214#issuecomment-810829250

There's one outstanding security warning for an outdated `node-fetch`, via `react-select` and `zooniverse-react-components`. Updating `prop-types` in ZRC might clear that but I'm not sure that it's a threat.

```
asosxlap127:pfe-lab odonnell$ npm why node-fetch
node-fetch@1.7.3
node_modules/node-fetch
  node-fetch@"^1.0.1" from isomorphic-fetch@2.2.1
  node_modules/isomorphic-fetch
    isomorphic-fetch@"^2.1.1" from fbjs@0.8.17
    node_modules/fbjs
      fbjs@"^0.8.9" from prop-types@15.5.10
      node_modules/zooniverse-react-components/node_modules/prop-types
        prop-types@"~15.5.10" from zooniverse-react-components@0.9.0
        node_modules/zooniverse-react-components
          zooniverse-react-components@"github:zooniverse/Zooniverse-React-Components#v0.9.1" from the root project
        prop-types@"^15.5.8" from react-select@1.0.0-rc.10
        node_modules/zooniverse-react-components/node_modules/react-select
          react-select@"1.0.0-rc.10" from zooniverse-react-components@0.9.0
          node_modules/zooniverse-react-components
            zooniverse-react-components@"github:zooniverse/Zooniverse-React-Components#v0.9.1" from the root project
```

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] Are the tests passing locally and on Travis?

### CODE REVIEW

File: .github/workflows/deploy_production.yml [hunk 1]
```diff
@@ -12,8 +12,19 @@ on:
...
+    env:
+      HEAD_COMMIT: ${{ github.sha }}
     steps:
-    - uses: actions/checkout@v2
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        persist-credentials: false
+
+    - name: Reconfigure git to use HTTP authentication
```
Comment (camallen) [inline .github/workflows/deploy_production.yml:23]:
```suggestion
    - name: Reconfigure git to use HTTPS protocol
```

File: .github/workflows/deploy_branch.yml [hunk 2]
```diff
@@ -8,7 +8,16 @@ jobs:
   deploy_staging_branch:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        persist-credentials: false
+
+    - name: Reconfigure git to use HTTP authentication
```
Comment (camallen) [inline .github/workflows/deploy_branch.yml:16]:
```suggestion
    - name: Reconfigure git to use HTTPS protocol
```

File: .github/workflows/deploy_staging.yml [hunk 3]
```diff
@@ -9,8 +9,19 @@ on:
...
+    env:
+      HEAD_COMMIT: ${{ github.sha }}
     steps:
-    - uses: actions/checkout@v2
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        persist-credentials: false
+
+    - name: Reconfigure git to use HTTP authentication
```
Comment (camallen) [inline .github/workflows/deploy_staging.yml:20]:
```suggestion
    - name: Reconfigure git to use HTTPS protocol
```

File: .github/workflows/run_tests.yml [hunk 4]
```diff
@@ -11,12 +11,29 @@ jobs:
   run_tests:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        persist-credentials: false
+
+    - name: Reconfigure git to use HTTP authentication
```
Comment (camallen) [inline .github/workflows/run_tests.yml:19]:
```suggestion
    - name: Reconfigure git to use HTTPS protocol
```

**Comment by eatyourgreens** [comment]
With the changes here, builds are updated to mirror the build process for PFE, security warnings from `npm audit` should be fixed and the latest commit ID will be published on deploy, so that we can check deployment status with lita.

**Review by camallen — APPROVED** [review]
LGTM

**Pull request merged**